src/main/java/org/myrobotlab/framework/Outbox.java

```diff
@@ -223,17 +232,32 @@ public void run() {
         }
         continue;
       }
-
     } // while (isRunning)
   }
+    
+  public FilterInterface addFilter(String name, String method, FilterInterface filter) {
+    return filters.put(String.format("%s.%s", CodecUtils.getFullName(name), method), filter);
+  }
+  
+  public FilterInterface removeFilter(String name, String method) {
+    return filters.remove(String.format("%s.%s", name, method));
+  }
+  
+  public boolean isFiltered(Message msg) {
+    if (filters.size() == 0 || !filters.containsKey(String.format("%s.%s", CodecUtils.getFullName(msg.name), msg.method))) {
+      return false;
+    } else {
+      return filters.get(String.format("%s.%s", msg.name, msg.method)).filter(msg);

```

Comment:
msg.name needs to be converted to full name like above. Probably be easiest to factor the string format above the conditional so it can be done once and used in both branches